resources/assets/lib/profile-page/account-standing.tsx

```diff
@@ -0,0 +1,145 @@
+          pattern={osu.trans('users.show.extra.account_standing.recent_infringements.actor')}
+        />
+      </span>
+    )}
+  </span>
+);
+
+const ColumnLength = ({ history }: ColumnProps) => {
+  if (history.type === 'restriction') {
+    return (
+      <div className={`${bn}__action ${bn}__action--restriction`}>
+        {osu.trans('users.show.extra.account_standing.recent_infringements.length_permanent')}
+      </div>
+    );
+  }
+
+  if (history.type === 'note') {
+    return null;
+  }
+
+  return <>{moment.duration(history.length, 'seconds').humanize()}</>;
+};
+
+const content: Record<Column, (props: ColumnProps) => JSX.Element | null> = {
+  action: ColumnAction,
+  date: ColumnDate,
+  description: ColumnDescription,
+  length: ColumnLength,
+};
+
+export default class AccountStanding extends React.PureComponent<ExtraPageProps> {
+  render() {
+    const latest = this.props.user.account_history.find((d) => d.type === 'silence');
+    const endTime = latest == null
+      ? null
+      : moment(latest.timestamp).add(latest.length, 'seconds');
+
+    return (
+      <div className='page-extra'>
+        <ExtraHeader name={this.props.name} withEdit={false} />
+
+        {latest != null && (
+          <div className='page-extra__alert page-extra__alert--warning'>
+            <StringWithComponent
+              mappings={{ username: <strong>{this.props.user.username}</strong> }}
+              pattern={stripTags(osu.trans('users.show.extra.account_standing.bad_standing'))}
+            />
+          </div>
+        )}
+
+        {endTime != null && endTime.isAfter() && (
+          <div className='page-extra__alert page-extra__alert--info'>
+            <StringWithComponent
+              mappings={{
+                duration: <TimeWithTooltip dateTime={endTime} relative />,
+                username: <strong>{this.props.user.username}</strong>,
+              }}
+              pattern={stripTags(osu.trans('users.show.extra.account_standing.remaining_silence'))}
+            />
+          </div>
+        )}
+
+        <h3 className='title title--page-extra-small'>
+          {osu.trans('users.show.extra.account_standing.recent_infringements.title')}
+        </h3>
+
+        <div className={bn}>
+          <table className={`${bn}__table`}>
+            <thead>
+              <tr>
+                {columns.map(this.renderHeaderColumn)}
+              </tr>
+            </thead>
+            <tbody>
+              {this.props.user.account_history.map((h) => (
+                <tr key={h.id}>
+                  {columns.map((column) => this.renderColumn(column, h))}
+                </tr>
+              ))}
+            </tbody>
+          </table>
+        </div>
+      </div>
+    );
+  }
+
+  private readonly renderColumn = (column: Column, history: UserAccountHistoryJson) => (
+    <td key={column} className={classWithModifiers(`${bn}__table-cell`, column)}>
+      {React.createElement(content[column], { history })}
```

Comment:
```
const Elem = content[column];
<Elem .... />
```
also works

Reply:
Well if you see the last comment :upside_down_face:

Reply:
🙃